### ISSUE TITLE
NVDA experiencing unexpected browse mode behavior with virtualized tables/grids

### ISSUE THREAD

**Issue by scottaohara** [nvaccess/nvda#13279]
The issue that we have found is that with large tables/grids which need to have virtualized columns and rows, when content is being removed from the table, navigating the table with NVDA's scan mode can lead to unexpected results.

### Steps to reproduce:
Go to https://codepen.io/hoicheng/full/BawKvYY or https://codepen.io/scottohara/pen/GRORORL
These reduced test cases demonstrates adding and removing columns to simulate what has been found to happen when navigating a larger table where columns and rows are removed for performance purposes.

### Actual behavior:
Navigating through the grid/table with scan mode, columns and rows are being announced as expected until content is add/removed.  In some cases, like when reaching the end of the table and content prior has been removed, table navigation keys will stop working and a user will have to use up/down arrow keys to move away from the cell they've become stuck on.

### Expected behavior:
As rows/cells become available in the table/grid, NVDA should be able to navigate to and away from them.  If rows/cells are removed from the table/grid, NVDA should not be prevented from navigating to or away from any cells which are available on screen and included in the browser's accessibility tree.

#### NVDA version:
2021.3

#### Windows version:
Win 10

#### Name and version of other software in use when reproducing the issue:
Edge  97.0.1072.69

#### Does the issue still occur after restarting your computer?
yes 

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
no

#### If NVDA add-ons are disabled, is your problem still occurring?
n/a

#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
n/a

**Comment by Adriani90** [comment]
Related or possibly duplicate of #8834. Though I think this test case here is easier to build, so I will let both issues open for now.
